src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs

```diff
@@ -689,49 +863,202 @@ private static IEnumerable<double> LazySequence(double records, double start, do
             }
         }
 
-        public static FormulaValue Abs(IRContext irContext, NumberValue[] args)
+        private static IEnumerable<decimal> LazySequenceDecimal(int records, decimal start, decimal step)
+        {
+            var x = start;
+            for (var i = 1; i <= records; i++)
+            {
+                yield return x;
+                x += step;
+            }
+        }
+
+        public static FormulaValue Abs(IRContext irContext, FormulaValue[] args)
         {
             var arg0 = args[0];
-            var x = arg0.Value;
-            var val = Math.Abs(x);
+
+            if (arg0 is NumberValue num)
+            {
+                return AbsFloat(irContext, num);
+            }
+            else if (arg0 is DecimalValue dec)
+            {
+                return AbsDecimal(irContext, dec);
+            }
+            else
+            {
+                return CommonErrors.UnreachableCodeError(irContext);
+            }
+        }
+
+        public static FormulaValue AbsFloat(IRContext irContext, NumberValue arg)
+        {
+            double x = arg.Value;
+            double val = Math.Abs(x);
             return new NumberValue(irContext, val);
         }
 
-        public static FormulaValue Round(IRContext irContext, NumberValue[] args)
+        public static FormulaValue AbsDecimal(IRContext irContext, DecimalValue arg)
+        {
+            decimal x = arg.Value;
+            decimal val = x < 0m ? -x : x;
+            return new DecimalValue(irContext, val);
+        }
+
+        public static FormulaValue Round(IRContext irContext, FormulaValue[] args)
         {
-            var numberArg = args[0].Value;
-            var digitsArg = args[1].Value;
+            double digits;
+
+            if (args.Length == 2 && args[1] is NumberValue numberDigs)
+            {
+                digits = numberDigs.Value;
+            }
+            else
+            {
+                return CommonErrors.UnreachableCodeError(irContext);
+            }
 
-            var x = Round(numberArg, digitsArg);
-            return new NumberValue(irContext, x);
+            if (args[0] is NumberValue num)
+            {
+                return RoundFloat(irContext, num, digits);
+            }
+            else if (args[0] is DecimalValue dec)
+            {
+                return RoundDecimal(irContext, dec, digits);
+            }
+
+            return CommonErrors.UnreachableCodeError(irContext);
         }
 
-        internal static double Round(double number, double digits, RoundType rt = RoundType.Default)
+        internal static FormulaValue RoundFloat(IRContext irContext, NumberValue num, double doubleDigs, RoundType rt = RoundType.Default)
         {
-            var s = number < 0 ? -1d : 1d;
+            var number = num.Value;
+            var s = number < 0 ? -1d : 1d;
             var n = number * s;

```

Comment:
>number * s [](http://example.com/codeflow?start=20&length=10)

double.Abs(number) ?

Reply:
This is what was there before, I just effectively added a line.  But it makes the code clearer to read, I changed it.  Note that it is Math.Abs() and it has an overload for Decimal too, so I'll fix that one up as well.